parquet-hadoop/src/main/java/parquet/hadoop/ParquetRecordReader.java

```diff
@@ -135,7 +135,18 @@ public void initialize(InputSplit inputSplit, TaskAttemptContext context)
               + context.getClass().getCanonicalName());
     }
 
-    initializeInternalReader((ParquetInputSplit)inputSplit, ContextUtil.getConfiguration(context));
+    // ensure all splits are ParquetInputSplits
+    ParquetInputSplit parquetSplit;
+    if (inputSplit instanceof ParquetInputSplit) {
+      parquetSplit = (ParquetInputSplit) inputSplit;
+    } else if (inputSplit instanceof FileSplit) {
+      parquetSplit = ParquetInputSplit.from((FileSplit) inputSplit);
+    } else {
+      throw new IllegalArgumentException(
+          "Invalid split (not a FileSplit or ParquetInputSplit): " + inputSplit);
+    }
+
+    initializeInternalReader(parquetSplit, ContextUtil.getConfiguration(context));
```

Comment:
I think this should be left as-is so we can start passing `FileSplit` instead of `ParquetInputSplit`. That would eliminate the need for Hive to do anything other than translate a mapred `FileSplit` to a mapreduce `FileSplit`. Then we can deprecate `ParquetInputSplit` from the public API and make it entirely internal and possibly remove it, once the client-side split strategy is gone.